scripts/irods/test/test_iput.py

```diff
@@ -886,6 +886,72 @@ def test_large_put(self):
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        # update the checksum_read_buffer_size_in_bytes to 10 MiB
+        svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = 10*1024*1024
+
+        # dump to a string to repave the existing server_config.json
+        new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+        try:
+            # create a 100 MiB file
+            file_size = 100*1024*1024
+            local_file = os.path.join(self.admin.local_session_dir, self.object_name)
+            lib.make_file(local_file, file_size)
+            file_checksum = lib.file_digest(local_file, 'sha256', encoding='base64')
+
+            with lib.file_backed_up(server_config_filename):
+                # repave the existing server_config.json
+                with open(server_config_filename, 'w') as f:
+                    f.write(new_server_config)
+
+                # put the file again with new server configuration and compare checksum
+                self.admin.assert_icommand(['iput', '-R', self.resource, '-k', local_file, self.object_name])
+                self.assertEqual(self.get_checksum(self.object_name).rstrip(), f'sha2:{file_checksum}')
+
+        finally:
+            # self.object_name deleted in teardown
+            os.unlink(local_file)
+
+    def test_server_rejects_invalid_server_config_options_for_checksum_read_buffer_size_in_bytes__issue_7947(self):
+        self.object_name = 'test_server_rejects_invalid_server_config_options_for_checksum_read_buffer_size_in_bytes__issue_7947'
+        server_config_filename = paths.server_config_path()
+
+        # load server_config.json to inject a new rule base
+        with open(server_config_filename) as f:
+            svr_cfg = json.load(f)
+
+        try:
+            # create a small file
+            file_size = 10
+            local_file = os.path.join(self.admin.local_session_dir, self.object_name)
+            lib.make_file(local_file, file_size)
+
+            with lib.file_backed_up(server_config_filename):
+
+                for buffer_size in [-2**63-1, -1, 0, 2**63]:  # first and last are outside range of 64 bit signed integers
+
+                    svr_cfg['advanced_settings']['checksum_read_buffer_size_in_bytes'] = buffer_size
+                    new_server_config = json.dumps(svr_cfg, sort_keys=True, indent=4, separators=(',', ': '))
+
+                    # repave the existing server_config.json
+                    with open(server_config_filename, 'w') as f:
+                        f.write(new_server_config)
+
+                    # put the file and verify we get an error due to invalid configuration
+                    self.admin.assert_icommand(['iput', '-f', '-R', self.resource, '-k', local_file, self.object_name], 'STDERR_SINGLELINE', 'CONFIGURATION_ERROR')
```

Comment:
The `-f` makes me itchy here. Maybe we should just `irm` the data object each iteration?

Reply:
The problem is that that class deletes self.object_name in its tearDown() method so if I delete every iteration that file will not exist to clean up.

Reply:
That's okay because `irm -f` doesn't produce an error in the event that the data object doesn't exist. Alternatively, you could wrap the try-finally with the loop, but I think it's better how it is now.

Speaking of which, each iteration of the loop could be made into a `unittest.subTest` if you're feeling up to it.

Reply:
I removed the -f and deleted the object on each iteration